modules/sdk-coin-avaxp/src/lib/atomicTransactionBuilder.ts

```diff
@@ -0,0 +1,192 @@
+    });
+
+    // validate the utxos
+    this.transaction._utxos.forEach((utxo) => {
+      if (!utxo) {
+        throw new BuildTransactionError('Utxo is undefined');
+      }
+      // addressesIndex should neve have a mismatch
+      if (utxo.addressesIndex?.includes(-1)) {
+        throw new BuildTransactionError('Addresses are inconsistent: ' + utxo.txid);
+      }
+      if (utxo.threshold !== this.transaction._threshold) {
+        throw new BuildTransactionError('Threshold is inconsistent');
+      }
+    });
+
+    this.transaction._utxos.forEach((utxo, i) => {
+      if (utxo.outputID === 7) {
+        const txidBuf = utils.cb58Decode(utxo.txid);
+        const amt: BN = new BN(utxo.amount);
+        const outputidx = utils.cb58Decode(utxo.outputidx);
+        const addressesIndex = utxo.addressesIndex ?? [];
+
+        // either user (0) or recovery (2)
+        const firstIndex = this.recoverSigner ? 2 : 0;
+        const bitgoIndex = 1;
+        currentTotal = currentTotal.add(amt);
+
+        const secpTransferInput = new SECPTransferInput(amt);
+
+        // if user/backup > bitgo
+        if (addressesIndex[bitgoIndex] < addressesIndex[firstIndex]) {
+          secpTransferInput.addSignatureIdx(addressesIndex[bitgoIndex], this.transaction._fromAddresses[bitgoIndex]);
+          secpTransferInput.addSignatureIdx(addressesIndex[firstIndex], this.transaction._fromAddresses[firstIndex]);
+          credentials.push(
+            SelectCredentialClass(
+              secpTransferInput.getCredentialID(), // 9
+              ['', this.transaction._fromAddresses[firstIndex].toString('hex')].map(utils.createSig)
+            )
+          );
+        } else {
+          secpTransferInput.addSignatureIdx(addressesIndex[firstIndex], this.transaction._fromAddresses[firstIndex]);
+          secpTransferInput.addSignatureIdx(addressesIndex[bitgoIndex], this.transaction._fromAddresses[bitgoIndex]);
+          credentials.push(
+            SelectCredentialClass(
+              secpTransferInput.getCredentialID(),
+              [this.transaction._fromAddresses[firstIndex].toString('hex'), ''].map(utils.createSig)
+            )
+          );
+        }
+
+        const input: TransferableInput = new TransferableInput(
+          txidBuf,
+          outputidx,
+          this.transaction._assetId,
+          secpTransferInput
+        );
+        inputs.push(input);
+      }
+    });
+
+    if (currentTotal.lt(totalTarget)) {
+      throw new BuildTransactionError(
+        `Utxo outputs get ${currentTotal.toString()} and ${totalTarget.toString()} is required`
+      );
+    } else if (currentTotal.gt(totalTarget)) {
+      outputs.push(
+        new TransferableOutput(
+          this.transaction._assetId,
+          new SECPTransferOutput(
+            currentTotal.sub(totalTarget),
+            this.transaction._fromAddresses,
+            this.transaction._locktime,
```

Comment:
this will be undefined as it's not set anywhere before

Reply:
Do you talk about locktime?

https://github.com/BitGo/BitGoJS/blob/7df888eca68bbc60505d837cdf8729c69f4d387e/modules/sdk-coin-avaxp/src/lib/transaction.ts#L67